docs/stm32/general.rst

```diff
@@ -0,0 +1,36 @@
+.. _stm32_general:
```

Comment:
This file should be called `docs/library/stm.rst` and be linked from `docs/library/index.rst` under "Libraries specific to the pyboard" section.

This file should also just be about this `stm` module, and doesn't need to describe the stm32 port in general.